src/main/java/io/cdap/plugin/aws/s3/source/S3BatchSource.java

```diff
@@ -202,6 +207,45 @@ public void validate(FailureCollector collector) {
       }
     }
 
+    void validatePath(FailureCollector collector) {
+      if (!containsMacro("path") && (!path.startsWith("s3a://") && !path.startsWith("s3n://"))) {
+        collector.addFailure("Path must start with s3a:// or s3n://.", null).withConfigProperty(NAME_PATH);
+      }
+      if (!containsMacro("path") && path.startsWith("s3n://") && !Strings.isNullOrEmpty(connection.getSessionToken())) {
+        collector.addFailure("Temporary credentials are only supported for s3a:// paths.", null)
+          .withConfigProperty(NAME_PATH);
+      }
+    }
+
+    public void validateCredentials(FailureCollector collector, String path) {
+      AmazonS3 s3 = getS3Client(Regions.DEFAULT_REGION.getName());
+      try {
+        String bucketRegion = s3.getBucketLocation(S3Path.from(path).getBucket());
+        s3 = getS3Client(bucketRegion);
+        s3.getBucketAcl(S3Path.from(path).getBucket());
+      } catch (AmazonS3Exception exception) {
+        String errorMessage = exception.getErrorMessage();
+        collector.addFailure(String.format("Invalid values: %s.", errorMessage),
```

Comment:
Please ensure that the ACL failures are tested using some creds which does not have ACL for the given path.

Reply:
As per our offline discussion, we're only validating the credentials and not the path, so reverted back to `listBuckets()` cf4e037c689aa95d587e91a4b47558de435a1d00

QA verified no issue observed with this API.